.travis.yml

```diff
@@ -2,128 +2,45 @@ sudo: false
-
-    # Lint
+        - CI_TYPES='cover goveralls'
     - go: 1.8
-      env: TYPE=lint
-
-    # Tests
-    #
-    # Need virtualenv for example tests
+      env:
+        - CI_TYPES='lint test'
     - go: 1.7
-      env: TYPE=test
-      addons: {apt: {packages: [python-virtualenv]}}
+      env:
+        - CI_TYPES=examples
+      addons:
+        apt:
+          packages:
+            - python-virtualenv
     - go: 1.8
-      env: type=test
-      addons: {apt: {packages: [python-virtualenv]}}
-    #- go: tip
-      #env: TYPE=test
-      #addons: {apt: {packages: [python-virtualenv]}}
+      env:
+        - CI_TYPES=examples
+      addons:
+        apt:
+          packages:
+            - python-virtualenv
 
 before_install:
-- go version
-- |
-  set -e
-  case "$TYPE" in
-    crossdock)
-      docker version
-      sudo rm -f /usr/local/bin/docker-compose
-      curl -L https://github.com/docker/compose/releases/download/${DOCKER_COMPOSE_VERSION}/docker-compose-`uname -s`-`uname -m` > docker-compose
-      chmod +x docker-compose
-      sudo mv docker-compose /usr/local/bin
-      docker-compose version
-      ;;
-  esac
-  set +e
-
+  - make ci-docker-load
 install:
-- |
-  set -e
-  make install
-  case "$TYPE" in
-    lint)
-      go get -u -f github.com/golang/lint/golint
-      ;;
-    test)
-      go get github.com/wadey/gocovmerge
-      go get github.com/mattn/goveralls
-      go get golang.org/x/tools/cmd/cover
-      ;;
-  esac
-  set +e
-
+  - make install
 script:
-- |
-  set -e
-  case "$TYPE" in
-    lint)
-      make lint
-      ;;
-    test)
-      rm -rf $GOPATH/pkg
-      make test_ci
-      make test-examples
-      travis_retry goveralls -coverprofile=cover.out -service=travis-ci || true
-      ;;
-    crossdock)
-      # We run this inside a subshell so that we can always emit
-      # `docker-compose logs` regardless of success or failure.
-      ( trap "docker-compose logs" EXIT; \
-        make crossdock )
-      ;;
-  esac
-  set +e
-
-after_success:
-- |
-  set -e
-  case "$TYPE" in
-    crossdock)
-      export REPO=yarpc/yarpc-go
-      export BRANCH=$(if [ "$TRAVIS_PULL_REQUEST" == "false" ]; then echo $TRAVIS_BRANCH; else echo $TRAVIS_PULL_REQUEST_BRANCH; fi)
-      export TAG=`if [ "$BRANCH" == "master" ]; then echo "latest"; else echo $BRANCH; fi`
-
-      docker build -f Dockerfile -t $REPO:$COMMIT .
-      docker tag $REPO:$COMMIT $REPO:$TAG
-      docker tag $REPO:$COMMIT $REPO:travis-$TRAVIS_BUILD_NUMBER
-      if [ "$CROSSDOCK_PUSH" == 1 ]; then
-        if [ -n "$DOCKER_EMAIL" ] && [ -n "$DOCKER_USER" ] && [ -n "$DOCKER_PASS" ]; then
-          docker login -e $DOCKER_EMAIL -u $DOCKER_USER -p $DOCKER_PASS
-          docker push $REPO
-        fi
-      fi
-      ;;
-  esac
-  set +e
+  - make ci
+  - if [[ "${TRAVIS_BRANCH}" == "master" ]] && [[ "${TRAVIS_PULL_REQUEST}" == "false" ]]; then make ci-docker-save; fi
```

Comment:
This isn't right. We want to push docker images for non-master branches too. The idea is that if you're developing a feature across two languages, you can consume the docker image for the feature branch from Go in the feature branch in Java.